src/allmydata/scripts/tahoe_invite.py

```diff
@@ -0,0 +1,109 @@
+import json
+from os.path import join
+from ConfigParser import NoSectionError
+
+from twisted.python import usage
+from twisted.internet import defer, reactor
+
+from wormhole import wormhole
+
+from allmydata.util import configutil
+from allmydata.util.encodingutil import argv_to_abspath
+from allmydata.scripts.common import get_default_nodedir
+
+
+class InviteOptions(usage.Options):
+    synopsis = "[options] <nickname>"
+    description = "Create a client-only Tahoe-LAFS node (no storage server)."
+
+    optParameters = [
+        ("shares-needed", None, None, "How many shares are needed to reconstruct files from this node"),
+        ("shares-happy", None, None, "Distinct storage servers new node will upload shares to"),
+        ("shares-total", None, None, "Total number of shares new node will upload"),
+    ]
+
+    def parseArgs(self, *args):
+        if len(args) != 1:
+            raise usage.UsageError(
+                "Provide a single argument: the new node's nickname"
+            )
+        self['nick'] = args[0].strip()
+
+
+@defer.inlineCallbacks
+def invite(options):
+    if options.parent['node-directory']:
+        basedir = argv_to_abspath(options.parent['node-directory'])
+    else:
+        basedir = get_default_nodedir()
+    config = configutil.get_config(join(basedir, 'tahoe.cfg'))
+    out = options.stdout
+    err = options.stderr
+
+    try:
+        introducer_furl = config.get('client', 'introducer.furl')
+    except NoSectionError:
+        # we're not a client; maybe this is running *on* the introducer?
```

Comment:
Yeah, maybe that would be better. There is a helper for that. I don't *really* care what type of node this is, it's more that there isn't just one way to get "the introducer fURL" -- maybe I really want to add a helper for that ...